### PR TITLE
Adds the ability to output projected band structure

### PR DESCRIPTION
**Pull request by 1041176461**
I still manually format the PDOS and projected band structure files in xml format, and do not use the Tinyxml-2 to format, because there are some confusion about using this code

### CODE REVIEW

File: source/src_io/energy_dos.cpp [hunk 1]
```diff
@@ -669,15 +642,286 @@ void energy::perform_dos(Local_Orbital_wfc &lowf, LCAO_Hamilt &uhm)
...
+			 }
+			 out <<std::endl<<std::endl;
+			 out <<std::setw(5)<< "io"<<std::setw(2)<<"="<<std::setw(2)<<"Orbital index in supercell"<<std::endl;
+			 out <<std::setw(5)<< "spec"<<std::setw(2)<<"="<<std::setw(2)<<"Atomic species label"<<std::endl;
+			 out <<std::setw(5)<< "l"<<std::setw(2)<<"="<<std::setw(2)<<"Angular mumentum quantum number"<<std::endl;
+			 out <<std::setw(5)<< "m"<<std::setw(2)<<"="<<std::setw(2)<<"Magnetic quantum number"<<std::endl;
+			 out <<std::setw(5)<< "z"<<std::setw(2)<<"="<<std::setw(2)<<"Zeta index of orbital"<<std::endl;
+			 out <<std::setw(5)<< "sym"<<std::setw(2)<<"="<<std::setw(2)<<"Symmetry name of real orbital"<<std::endl;
+			 out.close();}
+		}//out_band
```
Comment (dyzheng) [inline source/src_io/energy_dos.cpp:924]: Please format your code, the retraction looks wrong.

File: source/src_io/energy_dos.cpp [hunk 2]
```diff
@@ -580,7 +553,7 @@ void energy::perform_dos(Local_Orbital_wfc &lowf, LCAO_Hamilt &uhm)
 					 const int m1 = atom1->iw2m[j];
 					 out <<std::setw(5) << i << std::setw(8) 
 						<< GlobalC::ucell.atoms[t].label <<std::setw(5)
-							<<L1<<std::setw(5) <<m1<<std::setw(5)<<N1+1<<std::setw(15)<< Name_Angular[L1][m1] << std::endl;
+							<<L1<<std::setw(5) <<m1<<std::setw(5)<<N1+1<<std::setw(15)<< ModuleBase::Name_Angular[L1][m1] << std::endl;
```
Comment (dyzheng) [inline source/src_io/energy_dos.cpp:556]: Why do you move this label to module_base/constants.h?
If these labels only are only used in this file and you want to only define them once, you can move the code from constants.h to head of this file.
Will any other code use this labels in the future?

File: source/src_io/energy_dos.cpp (hunk 1 above)
Comment (1041176461) [reply source/src_io/energy_dos.cpp:924]: I've done the test and the program with this part of codes can be compiled and executed successfully

File: source/src_io/energy_dos.cpp (hunk 2 above)
Comment (1041176461) [reply source/src_io/energy_dos.cpp]: I moved this label from constant.h to src_pw/energy.h in the new commit

**Review by dyzheng — APPROVED** [review]

**Pull request merged**